src/app/search/autocomplete.ts

```diff
@@ -362,17 +362,20 @@ function findFilter<I, FilterCtx, SuggestionsCtx>(
   return filterName === 'is' ? filtersMap.isFilters[filterValue] : filtersMap.kvFilters[filterName];
 }
 
-// these filters might include quotes, so we search for two text segments to ignore quotes & colon
-// i.e. `name:test` can find `name:"test item"`
-const freeformTerms = freeformFilters.flatMap((f) => f.keywords).map((s) => `${s}:`);
-
 /**
  * This builds a filter-complete function that uses the given search config's keywords to
  * offer autocomplete suggestions for a partially typed term.
  */
 export function makeFilterComplete<I, FilterCtx, SuggestionsCtx>(
   searchConfig: SearchConfig<I, FilterCtx, SuggestionsCtx>,
 ) {
+  // these filters might include quotes, so we search for two text segments to ignore quotes & colon
+  // i.e. `name:test` can find `name:"test item"`
+  const freeformTerms = Object.values(searchConfig.filtersMap.kvFilters)
+    .filter((f) => canonicalFilterFormats(f.format).includes('freeform'))
+    .flatMap((f) => f.keywords)
+    .map((s) => `${s}:`);
```

Comment:
This is an actual change - instead of always using the freeform item filters, it now adapts to the filters passed into the function.